travisciapiprovider.py

```diff
@@ -9,22 +9,8 @@ class TravisCiApiProvider():
     log_url = host_url + '/jobs/{job_id}/log'
 
     def get_build(self, build_id):
-        return json.loads(urllib.urlopen(self.build_url.format(build_id=build_id)).read())
-
-
-    def _get_job_id(self, build_data, job_index=0):
-        try:
-            job_id = build_data['matrix'][job_index]['id']
-        except IndexError:
-            print "job_index out of bounds"
-            job_id = -1
-
-        return job_id
-
-
-    def get_log(self, build_data):
-        return urllib.urlopen(self.log_url.format(job_id=self._get_job_id(build_data))).read()
-
+        url = self.build_url.format(build_id=build_id)
+        return json.loads(urllib.urlopen(url).read())
```

Comment:
To support testing, this should use `self.get_page_content` instead of urllib directly (see https://github.com/servo/highfive/blob/master/newpr.py#L201).